src/black/trans.py

```diff
@@ -572,7 +575,20 @@ def make_naked(string: str, string_prefix: str) -> str:
         string_leaf = Leaf(token.STRING, S_leaf.value.replace(BREAK_MARK, ""))
 
         if atom_node is not None:
-            replace_child(atom_node, string_leaf)
+            # If not all children of the atom node are merged (this can happen
+            # when there is a standalone comment in the middle) ...
+            if non_string_idx - string_idx < len(atom_node.children):
+                # We need to replace the old STRING leaves with the new string leaf.
+                first_child_idx: Optional[int] = None
+                for idx in range(string_idx, non_string_idx):
+                    child_idx = LL[idx].remove()
+                    if first_child_idx is None:
+                        first_child_idx = child_idx
+                if first_child_idx is not None:
+                    atom_node.insert_child(first_child_idx, string_leaf)
```

Comment:
```suggestion
            if non_string_idx - string_idx < len(atom_node.children):
                # We need to replace the old STRING leaves with the new string leaf.
                first_child_idx = LL[string_idx].remove()
                for idx in range(string_idx + 1, non_string_idx):
                    LL[idx].remove()
                atom_node.insert_child(first_child_idx, string_leaf)
```
I'm assuming that `LL[string_idx]` always points to a valid string Leaf here, but this should be a lot clearer.

Reply:
great suggestion!

the `if first_child_idx is not None:` check is still needed for type checking, as `.remove()` returns an `Optional[int]`, though at runtime it shouldn't be `None`.

i updated with the latest commit.